src/vaadin-message-list.js

```diff
@@ -118,6 +131,75 @@ class MessageListElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     }
   }
 
+  /**
+   * @param {!KeyboardEvent} event
+   * @protected
+   */
+  _onKeydown(event) {
+    if (event.metaKey || event.ctrlKey) {
+      return;
+    }
+
+    // get index of the item that was focused when event happened
+    const target = event.composedPath()[0];
+    let currentIndex = this._messages.indexOf(target);
+
+    switch (event.key) {
+      case 'ArrowUp':
+        currentIndex--;
+        break;
+      case 'ArrowDown':
+        currentIndex++;
+        break;
+      case 'Home':
+        currentIndex = 0;
+        break;
+      case 'End':
+        currentIndex = this._messages.length - 1;
+        break;
+      default:
+        return; // nothing to do
+    }
+    if (currentIndex < 0) {
+      currentIndex = this._messages.length - 1;
+    }
+    if (currentIndex > this._messages.length - 1) {
+      currentIndex = 0;
+    }
+    this._focus(currentIndex);
+    event.preventDefault();
+  }
+
+  /**
+   * @param {number} idx
+   * @protected
+   */
+  _focus(idx) {
+    const target = this._messages[idx];
+    target.focus();
+  }
+
+  _handleFocusEvent(e) {
+    this._setTabIndexesByMessage(e.composedPath()[0]);
+  }
+  /**
+   * @param {number} idx
+   * @protected
+   */
+  _setTabIndexesByIndex(index) {
+    const message = this._messages[index] || this._messages[0];
+    this._setTabIndexesByMessage(message);
+  }
+
+  _setTabIndexesByMessage(message) {
+    this._messages.forEach((e) => (e.tabIndex = e === message ? 0 : -1));
+  }
```

Comment:
Should be tested: When focus moves from one message to another, the new focused message has tabindex 0 and all the others have -1.

Reply:
I added two tests, one for clicking and one for keyboard navigation.